pygmt/clib/session.py

```diff
@@ -56,6 +56,7 @@
     np.uint64: "GMT_ULONG",
     np.uint32: "GMT_UINT",
     np.datetime64: "GMT_DATETIME",
+    np.str_: "GMT_TEXT",
```

Comment:
Can we figure out if `GMT_TEXT` is the right enum here? Or should it be `GMT_CHAR` or something else?

Reply:
I don't think so. We even don't need this. Unlike GMT_Put_Vector, which can accept float, double or int, GMT_Put_Strings only accept strings. Thus we don't need to pass *gmt_type*.

Reply:
See this link for available GMT types: https://github.com/GenericMappingTools/gmt/blob/e4aab495643fb914b1a642cc6de4ffff041e1d62/src/gmt_resources.h#L84

Reply:
You're right. looking at the docs at https://docs.generic-mapping-tools.org/6.1/api.html#gmt-put-strings, it says:

>     int GMT_Put_Strings (void *API, unsigned int family, void *X, char **array);
> 
> where family is either GMT_IS_VECTOR or GMT_IS_MATRIX, X is either a GMT_VECTOR or GMT_MATRIX, and array is the a pointer to your string array.

Does that mean we need to pass in `GMT_VECTOR`/`GMT_MATRIX` instead of *gmt_type*? I'm confused as to how this works.

Reply:
We can attach the strings to a vector or a matrix. *X* can be a GMT_VECTOR or GMT_MATRIX, which is created by the `create_data` function. Thus, we have to pass *family* (`either GMT_IS_VECTOR or GMT_IS_MATRIX`) to let GMT know the type of *X*.